core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -422,9 +432,22 @@ class WorkspaceService(protected val userInfo: UserInfo,
       }
 
       // Delete Google Project
-      _ <- maybeDeleteGoogleProject(workspaceContext.googleProjectId, workspaceContext.workspaceVersion, userInfo)
+      _ <- maybeDeleteGoogleProject(workspaceContext.googleProjectId, workspaceContext.workspaceVersion, userInfo) recoverWith {
+        case t:Throwable => {
+          logger.error(s"Unexpected failure deleting workspace (while deleting google project) for workspace `${workspaceName}`", t)
+          throw t
+        }
+      }
+
+      // Delete the workspace records in Rawls. Do this after deleting the google project to prevent service perimeter leaks.
+      _ <- deleteWorkspaceTransaction(workspaceName, workspaceContext) recoverWith {
+        case t:Throwable => {
+          logger.error(s"Unexpected failure deleting workspace (while deleting workspace in Rawls DB) for workspace `${workspaceName}`", t)
+          throw t
```

Comment:
And then it'll get caught and logged again? What's the point of all this try then log stuff? If the answer is to have context, we'll have that already from the stack traces.

Reply:
I'm not a massive fan of this amount of log-then-throw stuff (since it'll probably get logged again further up the stack). But I won't drag hold you up about it.

Reply:
yeah that's understandable. it helps a lot with viewing what is causing issues with workspace deletion at a glance though! I think it's the lesser evil right now rather than having to dive into logs for every workspace deletion error to figure out what went wrong. (looking through doug's log sinks, most of the errors logged there do not have the stack trace bubbled up)